src/main/java/org/logstash/snmp/SnmpClientRequestAggregator.java

```diff
@@ -0,0 +1,175 @@
+                logger.info("failed to stop multi-thread request aggregator pool. Ignoring");
+            }
+        } catch (InterruptedException e) {
+            executor.shutdownNow();
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    public static class Request {
+        private final Executor executor;
+        private final SnmpClient client;
+        private final ConcurrentLinkedQueue<CompletableFuture<Map<String, ?>>> futures = new ConcurrentLinkedQueue<>();
+        private final Map<String, Object> result = new ConcurrentHashMap<>();
+
+        public Request(SnmpClient client, Executor executor) {
+            this.client = client;
+            this.executor = executor;
+        }
+
+        public void get(Target<Address> target, OID[] oids) {
+            submitRequestTask(target, "get", oids, Map.of(), () -> {
+                try {
+                    return client.get(target, oids);
+                } catch (IOException e) {
+                    throw new CompletionException(e);
+                }
+            });
+        }
+
+        public void walk(Target<Address> target, OID oid) {
+            submitRequestTask(target, "walk", new OID[]{oid}, Map.of(), () -> client.walk(target, oid));
+        }
+
+        public void table(Target<Address> target, String tableName, OID[] oids) {
+            submitRequestTask(target, "table", oids, Map.of("table_name", tableName), () -> client.table(target, tableName, oids));
+        }
+
+        private void submitRequestTask(Target<Address> target, String operation, OID[] oids, Map<String, String> extraLogDetails, Supplier<Map<String, ?>> task) {
+            final Supplier<Map<String, String>> logPropertiesSupplier = () -> createLogDetails(target, oids, extraLogDetails);
+
+            final CompletableFuture<Map<String, ?>> future = CompletableFuture.supplyAsync(task, executor)
+                    .exceptionally(ex -> handleRequestException(operation, ex, logPropertiesSupplier));
+
+            future.thenAccept(data -> handleRequestData(operation, data, logPropertiesSupplier));
+
+            this.futures.add(future);
+        }
+
+        private void handleRequestData(String operation, Map<String, ?> data, Supplier<Map<String, String>> logPropertiesSupplier) {
+            if (data != null && !data.isEmpty()) {
+                result.putAll(data);
+            } else if (data != null && logger.isDebugEnabled()) {
+                logger.debug("`{}` operation returned no response. {}", operation, logPropertiesSupplier.get());
+            }
+        }
+
+        private Map<String, ?> handleRequestException(String operation, Throwable ex, Supplier<Map<String, String>> logPropertiesSupplier) {
+            final Throwable throwable = isWrapperException(ex) && ex.getCause() != null ? ex.getCause() : ex;
+            logger.error("error invoking `{}` operation, ignoring. {}", operation, logPropertiesSupplier.get(), throwable);
```

Comment:
Looking at this in action:

```
[2024-04-24T12:44:57,557][ERROR][org.logstash.snmp.SnmpClientRequestAggregator][main][79044da4900979882d2aca9aa85f88350d9bc98e92a2ac0f9bbd5320f83c1a8a] error invoking `get` operation, ignoring. {host=127.0.0.1/1065, oids=[1.3.6.1.2.1.1.9.1.3.1]}
org.logstash.snmp.SnmpClientException: timeout sending snmp get request to target 127.0.0.1/1065
	at org.logstash.snmp.SnmpClient.get(SnmpClient.java:331) ~[logstash-integration-snmp-4.0.0.jar:?]
	at org.logstash.snmp.SnmpClientRequestAggregator$Request.lambda$get$0(SnmpClientRequestAggregator.java:97) ~[logstash-integration-snmp-4.0.0.jar:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:840) [?:?]
```

I'd suggest not having the stack trace unless it's in debug level.